lib/quirc.c

```diff
@@ -44,11 +49,33 @@ void quirc_destroy(struct quirc *q)
 	free(q);
 }
 
+
+int quirc_init(struct quirc* q, int w, int h, void* image)
+{
+	if (!q || !image || q->quirc_owns_buffers || !QUIRC_PIXEL_ALIAS_IMAGE) {
```

Comment:
I'll add a check on the width and height here similar to that in `quirc_resize()`, which has the following:

```
	/*
	 * XXX: w and h should be size_t (or at least unsigned) as negatives
	 * values would not make much sense. The downside is that it would break
	 * both the API and ABI. Thus, at the moment, let's just do a sanity
	 * check.
	 */
	if (w < 0 || h < 0)
		goto fail;
```

Perhaps those should be `<=`?

Reply:
IMO, from an API standpoint, the zero case is valid.

Reply:
Can you elaborate on how you see it as valid?  If the width or height is zero, then the image size must be zero bytes and there is no possibility of a QR code being detected.

Reply:
Obviously, I agree that no QR code are going to be detected in an empty image. However, detecting zero QR code is not an *error* but a *result*.

The empty image is a valid image just like the empty string is a valid string, or the empty array is a valid array etc. Robust API design simply accept the empty value for the domain. Imagine if an array search function would return an error or throw an exception when given the empty array.